### PR TITLE
feat(typeahead): add sort functionality to typeahead

### LINKED ISSUE #4808: Order results when using typeahead from ngx-bootstrap?
I am using customTemplate for my typeahead which has three fields

`({{model.number}}-{{model.city}}-{{model.state}})`

using the number as the typeaheadOptionField, I see there is typeaheadGroupField available which I use on the number field .... I would like to know is there anyway in which I can order the result based on the city field?

```
<ng-template #customCapTemplate let-model="item" let-index="index">
   <h5>{{model.code}}-{{model.city}}-{{model.prov}}</h5>
</ng-template>
<input name="colCap" [typeahead]="capList" typeaheadGroupField="code" 
    typeaheadOptionField="code" [typeaheadItemTemplate]="customCapTemplate" class="form-control">
```
screen shot 2018-11-15 at 12 43 20

### PR DESCRIPTION
**Pull request by IraErshova**
Add sort functionality to typeahead

Close #4808

### This PR depends on #5651

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated tests.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

**Review by daniloff200 — DISMISSED** [review]

File: src/typeahead/typeahead.directive.ts [hunk 1]
```diff
@@ -402,7 +408,7 @@ export class TypeaheadDirective implements OnInit, OnDestroy {
               );
           })
         )
-        .subscribe((matches: TypeaheadMatch[]) => {
+        .subscribe((matches: any[]) => {
```
Comment (daniloff200) [inline src/typeahead/typeahead.directive.ts:405]: according to chat with @IraErshova , there will be separate PR for fix of types in typeahead, because, this TypeaheadMatch type here was not exactly correct

Comment (IraErshova) [reply src/typeahead/typeahead.directive.ts:405]: Thanks for your comment!

Comment (IraErshova) [reply src/typeahead/typeahead.directive.ts:405]: [PR for typing](https://github.com/valor-software/ngx-bootstrap/pull/5647/commits/e21f68c18ce5d522f55ad13f798bfa08b3baca8d)

**Comment by dmitry-zhemchugov** [comment]
@IraErshova Needs clarification regarding the name of Demo and classes

**Comment by IraErshova** [comment]
> @IraErshova Needs clarification regarding the name of Demo and classes

@dmitry-zhemchugov I fixed this

**Review by daniloff200 — APPROVED** [review]
